app/script/conversation/ConversationRepository.js

```diff
@@ -339,23 +339,28 @@ z.conversation.ConversationRepository = class ConversationRepository {
       });
   }
 
-  get_conversations() {
-    const remote_conversations_promise = this.conversation_service.getAllConversations().catch(error => {
+  getConversations() {
+    const remoteConversationsSromise = this.conversation_service.getAllConversations().catch(error => {
       this.logger.error(`Failed to get all conversations from backend: ${error.message}`);
     });
 
-    return Promise.all([this.conversation_service.load_conversation_states_from_db(), remote_conversations_promise])
-      .then(([local_conversations, remote_conversations = []]) => {
-        if (remote_conversations.length) {
-          const conversations = this.conversationMapper.mergeConversation(local_conversations, remote_conversations);
-          return this.conversation_service.save_conversations_in_db(conversations);
+    return Promise.all([this.conversation_service.load_conversation_states_from_db(), remoteConversationsSromise])
+      .then(([localConversations, remoteConversations = []]) => {
+        const shouldMigrateData = localConversations.some(({muted_state}) => muted_state !== undefined);
+        if (shouldMigrateData) {
+          localConversations = this.conversationMapper.migrateConversationsData(localConversations, this.isTeam());
         }
```

Comment:
We check for the presence of some outdated data. Not sure if this is the best way to do it. It should only be true once once every machine. Maybe it is reliable enough to check the first object in the array instead of doing a some on the full array every time. WDYT?

Reply:
I like this solution. Two things:
`remoteConversationsSromise` => `remoteConversationsPromise`
`muted_state !== undefined` => `typeof muted_state !== 'undefined'`

Reply:
The second one should not pass the linter. Unless we have changed the rules: https://eslint.org/docs/rules/valid-typeof

Reply:
The only problem I can imagine with only checking the first item would be, if the migration crashes at some point, then we would have conversations that are never migrated.

It also depends on how often do we call this method. If it's only at app load time and then never again during the lifecycle of the app, I guess it's totally acceptable to do a `some`...

Reply:
Even if the array has thousands of items, we are talking about a < 10ms optimization I believe.
I think I would go for the safer solution here that could prevent very unexpected bugs